pkg/model/api/v1/datasource/http.go

```diff
@@ -70,49 +71,80 @@ func (h *HTTPAccess) validate() error {
 }
 
 type HTTPAllowedEndpoint struct {
-	Endpoint string `json:"endpoint" yaml:"endpoint"`
-	Method   string `json:"method" yaml:"method"`
+	EndpointPattern *regexp.Regexp `json:"endpoint_pattern" yaml:"endpoint_pattern"`
+	Method          string         `json:"method" yaml:"method"`
+}
+
+type tmpHTTPAllowedEndpoint struct {
```

Comment:
I kind of like this approach to making regex fields YAML-marshalable: https://github.com/prometheus/prometheus/blob/60d099088692b6b5bf4b8f4e13f76bd3921da99e/pkg/relabel/relabel.go#L135-L180

And then it's just used like this: https://github.com/prometheus/prometheus/blob/60d099088692b6b5bf4b8f4e13f76bd3921da99e/pkg/relabel/relabel.go#L80-L81

Then it can also be reused easily in other places if defined like that. What do you think?

Reply:
ah yeah that's definitely better and smarter. Thanks !